_changes.html

```diff
@@ -12442,8 +12442,354 @@
+<ul class="bugfixes">
+ BGF Made the progress meter use HHH:MM:SS instead of only seconds.
+</ul>
+
+<a name="2_4"></a>
+SUBTITLE(Fixed in 2.4 - August 27 1997 )
+<p> Changes:
+<ul class="changes">
+ CHG Added progress meter. It appears when downloading > BUFFER SIZE and
+   mute is not selected. I found out that when downloading large files from
+   really really slow sites, it is desirable to know the status of the
+   download. Do note that some downloads are done unawaring of the size, which
+   makes the progress meter less thrilling ;) If the output is sent to a tty,
+ the progress meter is shut off.
+ CHG Made it grok user+passwd for HTTP fetches. The trick is to base64
+   encode the user+passwd and send an extra header line. Read chapter 11.1 in
+   RFC2068 for details. I added it to be used just like the ftp one.
+</ul>
+
+<p> Bugfixes:
+<ul class="bugfixes">
+ BGF Increased buffer size used for reading.
+ BGF Added length checks in the user+passwd parsing.
+</ul>
+
+<a name="2_3"></a>
+SUBTITLE(Fixed in 2.3 - August 21 1997 )
+<p> Changes:
+<ul class="changes">
+ CHG Added "-o" option (output file)
+ CHG Added URG_HTTP_NOT_FOUND return code.
+ CHG Looping mode ("-l" option). It's easier to get various split files.
+ CHG Changed the -h to be -i instead. -h should be preserved to help use.
+</ul>
+<p> Bugfixes:
+<ul class="bugfixes">
+ BGF Bjorn Reese indicated that Borland _might_ use '_WIN32' instead of the
+   VC++ WIN32 define and therefore I added a little fix for that.
+</ul>
+
+<a name="2_2"></a>
+SUBTITLE(Fixed in 2.2 - August 14 1997 )
+<p> Changes:
+<ul class="changes">
+ CHG Made a '-h' switch in case you want the HTTP-header in the output.
+</ul>
+<p> Bugfixes:
+<ul class="bugfixes">
+ BGF The urlget function didn't set the path to url when using proxy.
+ BGF Fixed bug with IMC proxy. Now using (almost) complete GET command.
+ BGF Made it compile on Solaris. Had to reorganize the includes a bit.
+   (so Win32, Linux, SunOS 4 and Solaris 2 compile fine.)
+ BGF Made Johan's keepalive keyword optional with the -k flag (since it
+   makes a lot of urlgets take a lot longer time).
+</ul>
+
+<a name="2_1"></a>
+SUBTITLE(Fixed in 2.1 - August 10 1997 )
+<p> Changes:
+<ul class="changes">
+ CHG Mute option (no output at all to stderr)
+</ul>
+<p> Bugfixes:
+<ul class="bugfixes">
+ BGF Win32-compilable
+ BGF No more global variables
+ BGF Full range of return codes from urlget(), which is now written to be a
+   function for easy-to-use in [other] programs.
+ BGF Define STAND_ALONE to compile the stand alone urlget program
+ BGF Now compiles with gcc options -ansi -Wall -pedantic ;)
+</ul>
+
+<a name="2_0"></a>
+SUBTITLE(Fixed in 2.0 - August 1 1997 )
+<ul class="changes">
+ CHG Introducing ftp GET support. The FTP URL type is recognized and used.
+ CHG Renamed the project to <b>urlget</b>
+ CHG Supports the user+passwd in the FTP URL (otherwise it tries anonymous
+   login with a weird email address as password).
+</ul>
+
+<hr>
```

Comment:
Noticed on the [website](https://curl.se/changes.html) that separator `<hr>` is only used between 1.5 and 2.0 and between 3.12 and 4.0. The horizontal ruler is not present between:

- 0.3 and 1.0
- 2.9 and 3.0
- 4.10 and 5.0
- 5.11 and 6.0
- 6.5.2 and 7.1 (sidenote: where is 7.0?)

Reply:
I put the `<hr>` a little half-assed where there was a rename: when httpget become urlget and when urlget become curl. We could probably come up with something better to visualized that.

There was no 7.0 release. We did a series of beta releases and then we bumped to 7.1 to avoid any confusions. You can see that there are a few other gaps in the release train as well.